### DISCUSSION TITLE
hi ☺️ Generic help request

### DISCUSSION THREAD

**Discussion by kylprq** [HotCakeX/Harden-Windows-Security#33] [Q&A] [answered]
I'm getting " Get-WMIObject : The term 'Get-WMIObject' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. " error.
Thank you for your time have great day 👩🏼‍💻

**Comment by HotCakeX** [comment, marked as answer]
Hi,
The hardening script no longer uses `Get-WMIObject` and instead uses the more modern `Get-CimInstance`.

https://learn.microsoft.com/en-us/powershell/module/cimcmdlets/get-ciminstance?view=powershell-5.1

Could you explain more about the scenario where you get that error? Like OS version, PowerShell version and anything else that can help, because I can't reproduce it on my side.

You too!

**Reply by kylprq** [reply]
![pwsh_KCTweNuDN3](https://user-images.githubusercontent.com/127903002/235587075-9bdd9eed-5fc6-43d2-9426-b5d2730a32bc.gif)

https://i.imgur.com/wuTdAkg.gif

hope it helps! 🐱‍👤

**Reply by HotCakeX** [reply]
Hi, thank you for the gif,
Here is an animated apng i recorded: https://i.postimg.cc/jsvpn5nq/1.png

What the script does is essentially searching for the currently running PowerShell executables, adding them to Controlled Folder Access exclusions and then at the end or whenever user exits the script execution, they will be removed from exclusion list.

Before rolling out that feature, I tested it a Lot to make sure everything is working, but I couldn't and still can't reproduce the issues you showed me, i tried multiple times again.

**Can you confirm a few things for me please,**

1. Are you using latest Windows 11 version?
2. Have you applied any configurations on your system that might interfere somehow with the normal workflow of Windows and/or PowerShell?
3. Show me the output of `"$PSHOME\*.exe"`
4. Do you get the errors on Windows PowerShell too?
5. Are you using any AV solution other than Microsoft Defender?
6. Is your Windows image tampered or modified in an unofficial way?

There is only 1 instance of WMI being used and it's by [Microsoft Security baselines script](https://www.microsoft.com/en-us/download/details.aspx?id=55319) just to check the OS is not a Windows Server domain controller.

```powershell
((Get-WmiObject Win32_OperatingSystem).ProductType -eq 2)
```

**Reply by kylprq** [reply]
Ok I'll try post back. Huge thanks for your efforts 🐱‍💻

**Reply by HotCakeX** [reply]
Thank you 🙏😊

**Comment by kylprq** [comment]
can be ignored (problem is on my end🤕)🤷🏽‍♀️
in any case thank you for your efforts to keep people safe🌌
